astropy/coordinates/representation.py

```diff
@@ -538,12 +773,24 @@ def _combine_operation(self, op, other, reverse=False):
             Whether the operands should be reversed (e.g., as we got here via
             ``self.__rsub__`` because ``self`` is a subclass of ``other``).
         """
+        self._raise_if_has_differentials(op.__name__)
+
         result = self.to_cartesian()._combine_operation(op, other, reverse)
         if result is NotImplemented:
             return NotImplemented
         else:
             return self.from_cartesian(result)
 
+    # We need to override this setter to support differentials
+    @BaseRepresentationOrDifferential.shape.setter
+    def shape(self, shape):
+        # See: https://stackoverflow.com/questions/3336767/ for an example
+        BaseRepresentationOrDifferential.shape.fset(self, shape)
```

Comment:
If `with_differentials` is still the version that makes a copy, then indeed it should work -- no strides of 0 any more. I haven't installed your branch yet, but would expect that
```
rep = r.CartesianRepresentation(*xyz, differentials=dif)
rep.shape = (10,)
```
would fail, with `rep.shape` already changed to `(10,)`.